docs/Talkbuchet.js

```diff
@@ -0,0 +1,609 @@
+ *   reload the page and copy and paste the script again.
+ *
+ * HOW TO ENABLE AND DISABLE THE MEDIA DURING A TEST:
+ * -----------------------------------------------------------------------------
+ * You can manually enable and disable the media during a test by copying and
+ * pasting in the browser console the following commands:
+ * - For audio:
+ * stream.getAudioTracks()[0].enabled = TRUE_OR_FALSE
+ * - For video:
+ * stream.getVideoTracks()[0].enabled = TRUE_OR_FALSE
+ *
+ * Note that you can only enable and disable the original media specified in the
+ * "getUserMedia" call.
+ *
+ * HOW TO CALIBRATE:
+ * -----------------------------------------------------------------------------
+ * The script starts as many publishers and subscripters for each publisher as
+ * specified, each one being a real and independent peer connection to Janus.
+ * Therefore, how many peer connections can be established with a single script
+ * run depends on the client machine, both its CPU and its network.
+ *
+ * It is first needed to "calibrate" the client by running it with different
+ * numbers of publishers and subscribers to check how many peer connections are
+ * supported before doing the actual stress test on Janus.
+ *
+ * It is most likely that the CPU or network of a single client will be
+ * saturated before Janus is. The script will write to the console when a peer
+ * could not be connected or if it was disconnected after being connected; if
+ * there is a large number of those messages or the CPU consumption is very high
+ * the client has probably reached its limit.
+ *
+ * Besides the messages written by the script itself you can manually check the
+ * connection state by copying and pasting in the browser console the following
+ * commands:
+ * - For the publishers:
+ * Object.values(publishers).forEach(publisher => { console.log(publisher.peerConnection.iceConnectionState) })
+ * - For the subscribers:
+ * subscribers.forEach(subscriber => { console.log(subscriber.peerConnection.iceConnectionState) })
+ *
+ * DISCLAIMER:
+ * -----------------------------------------------------------------------------
+ * Talk performs some optimizations during calls, like reducing the video
+ * quality based on the number of participants. This script does not take into
+ * account those things; it is meant to be used for stress tests of Janus rather
+ * than to accurately simulate Talk behaviour.
+ *
+ * Independently of that, please note that an accurate simulation would not be
+ * possible given that a single client has to behave as several different
+ * clients. In a real call each client could have a different behaviour (not
+ * only due to sending different media streams, but also from having different
+ * CPU and network conditions), and that might even also affect Janus and the
+ * server regarding very low level things (but relevant for performance on
+ * highly loaded systems) like caches.
```

Comment:
note for later: try avoiding using parenthesis too often, it breaks the flow of sentences. Try embedding them directly into sentences. :-)

Reply:
Feel free to rewrite it. I am perfectly aware that I abuse parenthesis (even if they break the flow of sentences :-P :-P :-P).